src/CivicrmEntityViewsData.php

```diff
@@ -382,12 +383,17 @@ protected function processViewsDataForSpecialFields(array &$views_field, $base_t
         break;
 
       case 'civicrm_contact':
+        $user_definition = $this->entityTypeManager->getDefinition('user');
+        assert($user_definition !== NULL);
+        $drupal_connection = Database::getConnectionInfo();
+        $database = $drupal_connection['default']['database'];
         $views_field['civicrm_contact']['user'] = [
           'title' => $this->t('User related to the CiviCRM contact'),
           'help' => $this->t('Relate user to the CiviCRM contact.'),
           'relationship' => [
-            'base' => 'users_field_data',
-            'base field' => 'uid',
+            // @todo: add the Drupal database name as a prefix `db.user_field_data`.
+            'base' => $database . '.' . $user_definition->getDataTable(),
+            'base field' => $user_definition->getKey('id'),
```

Comment:
This should work. But locally, on my machine, it keeps picking the `db.users_field_data` which is not the prefixed one from the test. So I'm guessing that adding the database name before the table breaks prefixing in Drupal. So, back to the drawing board a little bit.

I have to run. The fix is probably adding the prefix as well.